packages/ifpack2/src/Ifpack2_LinePartitioner_def.hpp

```diff
@@ -139,18 +139,18 @@ int LinePartitioner<GraphType,Scalar>::Compute_Blocks_AutoLine(Teuchos::ArrayVie
 
     // Get neighbors and sort by distance
     this->Graph_->getLocalRowCopy(i,cols(),nz);
-    Scalar x0 = (!xvals.is_null()) ? xvals[i/NumEqns_] : zero;
-    Scalar y0 = (!yvals.is_null()) ? yvals[i/NumEqns_] : zero;
-    Scalar z0 = (!zvals.is_null()) ? zvals[i/NumEqns_] : zero;
+    double x0 = (!xvals.is_null()) ? xvals[i/NumEqns_] : zero;
+    double y0 = (!yvals.is_null()) ? yvals[i/NumEqns_] : zero;
+    double z0 = (!zvals.is_null()) ? zvals[i/NumEqns_] : zero;
 
     LO neighbor_len=0;
     for(size_t j=0; j<nz; j+=NumEqns_) {
       MT mydist = mzero;
```

Comment:
If MT comes from Scalar, MT could be float instead of double.

Either use double everywhere, or make a typedef for the coordinate type and use that consistently.